subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -89,15 +109,60 @@ int nrf_cloud_init(const struct nrf_cloud_init_param *param)
 	return 0;
 }
 
-int nrf_cloud_connect(const struct nrf_cloud_connect_param *param)
+static int connect_error_translate(const int err)
 {
-	if (NOT_VALID_STATE(STATE_INITIALIZED)) {
-		return -EACCES;
+	switch (err) {
+	case 0:
+		return NRF_CLOUD_CONNECT_RES_SUCCESS;
+	case -ECHILD:
+		return NRF_CLOUD_CONNECT_RES_ERR_NETWORK;
+	case -EACCES:
+		return NRF_CLOUD_CONNECT_RES_ERR_NOT_INITD;
+	case -ENOEXEC:
+		return NRF_CLOUD_CONNECT_RES_ERR_BACKEND;
+	case -EINVAL:
+		return NRF_CLOUD_CONNECT_RES_ERR_PRV_KEY;
+	case -EOPNOTSUPP:
+		return NRF_CLOUD_CONNECT_RES_ERR_CERT;
+	case -ECONNREFUSED:
+		return NRF_CLOUD_CONNECT_RES_ERR_CERT_MISC;
+	case -ETIMEDOUT:
+		return NRF_CLOUD_CONNECT_RES_ERR_TIMEOUT_NO_DATA;
+	case -ENOMEM:
+		return NRF_CLOUD_CONNECT_RES_ERR_NO_MEM;
+	case -EINPROGRESS:
+		return NRF_CLOUD_CONNECT_RES_ERR_ALREADY_CONNECTED;
+	default:
+		LOG_ERR("nRF cloud connect failed %d", err);
+		return NRF_CLOUD_CONNECT_RES_ERR_MISC;
 	}
+}
+
+static int connect_to_cloud(void)
+{
 	atomic_set(&disconnect_requested, 0);
 	return nct_connect();
 }
 
+int nrf_cloud_connect(const struct nrf_cloud_connect_param *param)
+{
+	if (NOT_VALID_STATE(STATE_INITIALIZED)) {
+		return NRF_CLOUD_CONNECT_RES_ERR_NOT_INITD;
+	}
+
+	int err;
```

Comment:
Maybe place variable definition in the top of the function. I'm not strongly opinionated on this but it would align with the rest of the functions in this library.